src/TLuaInterpreter.cpp

```diff
@@ -944,31 +944,49 @@ int TLuaInterpreter::selectCaptureGroup(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getLines
 int TLuaInterpreter::getLines(lua_State* L)
 {
-    int luaFrom;
-    if (!lua_isnumber(L, 1)) {
-        lua_pushfstring(L, "getLines: bad argument #1 type (starting line to get as number expected, got %s!)", luaL_typename(L, 1));
+    int n = lua_gettop(L);
+    int s = 0;
+    QString windowName = QLatin1String("main");
```

Comment:
```suggestion
    QString windowName = QStringLiteral("main");
```

Reply:
There ***will*** be many of these in the near future at least in this compilation unit so the non-compile time creation cost from using the thin wrapper that is `QLatin1String` will be shared across all the instances within that file if I understand it correctly. The `QString` constructor does have a `QLatin1String` overload so this is should not degenerate into something that is as bad as a raw `QString("main")`...

If the string concerned was only used once or twice in the file I agree that `QStringLiteral` is the one to use, but this is an exception that proves the rule IMVHO.